### ISSUE TITLE
Nextcloud won't start after Upgrade

### ISSUE THREAD

**Issue by david-steg** [nextcloud/all-in-one#1938]
Hello After an Major Upgrade, my Nextcloud Container cannot fully Start.
Please Help. Thank You in Advance!

### Steps to reproduce
Upgrade from AIO 1.4 to 4.2
Resize Linux Partition, due to space shortage.
Upgrade Container, Version unknowns

### Expected behavior 
Start all Container.

### Actual behavior 
Netcloud is stuck with Status "Starting"

### Host OS 
Ubuntu 20.04

#### Nextcloud AIO version 
4.2.0

#### Other valuable info 
Logs from Netcloud (The last part):
```
              now              
-------------------------------
 2023-02-05 15:12:32.998877+00
(1 row)

+ '[' -f /dev-dri-group-was-added ']'
++ find /dev -maxdepth 1 -mindepth 1 -name dri
+ '[' -n '' ']'
+ set +x
Configuring Redis as session handler...
Setting php max children...
+ NEXT_MAJOR=24
+ curl -fsSL -o nextcloud.tar.bz2 https://download.nextcloud.com/server/releases/latest-24.tar.bz2
+ curl -fsSL -o nextcloud.tar.bz2.asc https://download.nextcloud.com/server/releases/latest-24.tar.bz2.asc

p -d
+ GNUPGHOME=/tmp/tmp.dGpRKNNzn2
+ export GNUPGHOME
+ gpg --batch --keyserver keyserver.ubuntu.com --recv-keys ***
gpg: keybox '/tmp/tmp.dGpRKNNzn2/pubring.kbx' created
gpg: /tmp/tmp.dGpRKNNzn2/trustdb.gpg: trustdb created
gpg: key ***: public key "Nextcloud Security <security@nextcloud.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
+ gpg --batch --verify nextcloud.tar.bz2.asc nextcloud.tar.bz2
gpg: Signature made Tue Jan 17 09:15:49 2023 UTC
gpg:                using RSA key ***
gpg: Good signature from "Nextcloud Security <security@nextcloud.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: ***
+ mkdir -p /usr/src/tmp
+ tar -xjf nextcloud.tar.bz2 -C /usr/src/tmp/
+ gpgconf --kill all
+ rm nextcloud.tar.bz2.asc nextcloud.tar.bz2
+ mkdir -p /usr/src/tmp/nextcloud/data
+ mkdir -p /usr/src/tmp/nextcloud/custom_apps
+ chmod +x /usr/src/tmp/nextcloud/occ
+ cp /usr/src/nextcloud/config/CAN_INSTALL /usr/src/nextcloud/config/apcu.config.php /usr/src/nextcloud/config/apps.config.php /usr/src/nextcloud/config/config.sample.php /usr/src/nextcloud/config/redis.config.php /usr/src/nextcloud/config/reverse-proxy.config.php /usr/src/nextcloud/config/s3.config.php /usr/src/nextcloud/config/swift.config.php /usr/src/tmp/nextcloud/config/
+ mkdir -p /usr/src/tmp/nextcloud/apps/nextcloud-aio
+ cp /usr/src/nextcloud/apps/nextcloud-aio/appinfo /usr/src/nextcloud/apps/nextcloud-aio/composer /usr/src/nextcloud/apps/nextcloud-aio/lib /usr/src/nextcloud/apps/nextcloud-aio/readme.md /usr/src/nextcloud/apps/nextcloud-aio/templates /usr/src/tmp/nextcloud/apps/nextcloud-aio/
cp: -r not specified; omitting directory '/usr/src/nextcloud/apps/nextcloud-aio/appinfo'
cp: -r not specified; omitting directory '/usr/src/nextcloud/apps/nextcloud-aio/composer'
cp: -r not specified; omitting directory '/usr/src/nextcloud/apps/nextcloud-aio/lib'
cp: -r not specified; omitting directory '/usr/src/nextcloud/apps/nextcloud-aio/templates'
              now              
-------------------------------
 2023-02-05 15:13:42.008883+00
(1 row)

+ '[' -f /dev-dri-group-was-added ']'
++ find /dev -maxdepth 1 -mindepth 1 -name dri
+ '[' -n '' ']'
+ set +x
Configuring Redis as session handler...
Setting php max children...
+ NEXT_MAJOR=24
+ curl -fsSL -o nextcloud.tar.bz2 https://download.nextcloud.com/server/releases/latest-24.tar.bz2
+ curl -fsSL -o nextcloud.tar.bz2.asc https://download.nextcloud.com/server/releases/latest-24.tar.bz2.asc

p -d
+ GNUPGHOME=/tmp/tmp.B3ge12nW2Q
+ export GNUPGHOME
+ gpg --batch --keyserver keyserver.ubuntu.com --recv-keys ***
gpg: keybox '/tmp/tmp.B3ge12nW2Q/pubring.kbx' created
gpg: /tmp/tmp.B3ge12nW2Q/trustdb.gpg: trustdb created
gpg: key ***: public key "Nextcloud Security <security@nextcloud.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
+ gpg --batch --verify nextcloud.tar.bz2.asc nextcloud.tar.bz2
gpg: Signature made Tue Jan 17 09:15:49 2023 UTC
gpg:                using RSA key ***
gpg: Good signature from "Nextcloud Security <security@nextcloud.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: ***
+ mkdir -p /usr/src/tmp
+ tar -xjf nextcloud.tar.bz2 -C /usr/src/tmp/
```

**Comment by szaimen** [comment]
Thanks for the report! Looks like there is indeed a problem when upgrading beftween multiple major versions. I take a look at this tomorrow and will try to fix this in the next beta version.

Please restore a backup until then!

PS: Seems like you are the first person that ran into this because you didnt upgrade at least once in the past 6 months. Congrats!

**Comment by david-steg** [comment]
Hello, Thank you for the quick response!
I already did a Rollback, so I will wait until the next release. No Pressure ;)
Yes... I wasn't very diligently about Nextcloud in the last few Months.. :/

**Comment by szaimen** [comment]
This is now fixed with v4.3.2 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by david-steg** [comment]
Hello Simon
Thank you for the quick fix, very impressiv!
The Upgrade with the Beta went with no Problem! No Problem so far.

Juste about the readme.md, if I might...
The Chapiter "How to Switch the Chanel?" is not Dummy-proof, like for me ;).
1. I wasn't sure if the Docker-Image Repository have to be change inside the AIO Interface or in my Docker-Compose File, at the end I just tried to change it inside the Docker Compose file, fortunately it works. Maybe a small indication about this would be nice.
2. You talk about "stopping the mastercontainer, removing it", what thos it mean? Delete the Container? Image? Volume? I wasn't sure, first I dummly Delete the Volume, it resetet everything, that wasn't the point.
At the second try (after a Restore), I just tried to delete the Container, but I coun't find any mastercontainer. So I just did a "docker compose up -d" and it just worked.

Maybe it's just me, but I needed a few trys, fortunately I work in a VM. 
You asked about a Feedback, here you have it, hopfully it helps. ;)
